src/huggingface_hub/inference/_client.py

```diff
@@ -109,6 +111,12 @@
 MODEL_KWARGS_NOT_USED_REGEX = re.compile(r"The following `model_kwargs` are not used by the model: \[(.*?)\]")
 
 
+@dataclass
+class _InferenceInputs:
+    json: Optional[Dict[str, Any]] = None
+    raw_data: Optional[ContentT] = None
```

Comment:
Potentially slightly overkill. No need to be extremely precise on the return type of `_prepare_payload`. You can also add `-> Dict` as return type with output values like `{"payload": ...}` or `{"data": ...}` (it's never both at the same time).

And then call it like this:

```py
response = self.post(
    **_prepare_payload(inputs, parameters),
    model=model,
    task="document-question-answering",
)
```